src/OSPSuite.Presentation/Presenters/Charts/CurveSettingsPresenter.cs

```diff
@@ -152,35 +148,34 @@ public void NotifyCurvePropertyChange(CurveDTO curveDTO)
          CurvePropertyChanged(curveDTO.Curve);
       }
 
-      public event Action<Curve> CurvePropertyChanged = delegate { };
-
       private void rebind()
       {
          View.BindTo(_allCurvesDTOs);
       }
 
-      private void addCurve(Curve curve)
-      {
-         _allCurvesDTOs.Add(mapFrom(curve));
-      }
+      private void addCurve(Curve curve) => _allCurvesDTOs.Add(mapFrom(curve));
 
       private bool hasCurveDTOFor(Curve curve)
       {
          return _allCurvesDTOs.Any(x => Equals(x.Curve, curve));
       }
 
-
-      public IEnumerable<AxisTypes> AllYAxisTypes => _chart.AllUsedYAxisTypes;
-
-      public void SetCurveXData(CurveDTO curve, DataColumn dataColumn)
+      public void SetCurveXData(CurveDTO curveDTO, DataColumn dataColumn)
       {
-         _chart.SetxData(curve.Curve, dataColumn, _dimensionFactory);
+         _chart.SetxData(curveDTO.Curve, dataColumn, _dimensionFactory);
+         NotifyCurvePropertyChange(curveDTO);
       }
 
-      public void SetCurveYData(CurveDTO curve, DataColumn dataColumn)
+      public void SetCurveYData(CurveDTO curveDTO, DataColumn dataColumn)
       {
-         _chart.SetyData(curve.Curve, dataColumn, _dimensionFactory);
+         _chart.SetyData(curveDTO.Curve, dataColumn, _dimensionFactory);
+         NotifyCurvePropertyChange(curveDTO);
       }
 
+      public void UpdateCurveColor(CurveDTO curveDTO, Color color)
+      {
```

Comment:
that was done in the View directly. Another example of view that should not do anything